src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/DataSetService.cs

```diff
@@ -108,4 +133,66 @@ private static DataSetViewModel MapDataSet(DataSet dataSet)
             .AsNoTracking()
             .FirstOrNotFoundAsync(p => p.Id == publicationId, cancellationToken: cancellationToken);
     }
+
+    private async Task<Either<ActionResult, ReleaseVersion>> CheckReleaseVersionExists(
+        Guid releaseVersionId,
+        CancellationToken cancellationToken = default)
+    {
+        return await contentDbContext.ReleaseVersions
+            .AsNoTracking()
+            .SingleOrNotFoundAsync(rv => rv.Id == releaseVersionId, cancellationToken: cancellationToken);
+    }
+
+    private async Task<Either<ActionResult, DataSetVersion>> CheckVersionExists(
+        Guid dataSetId,
+        string dataSetVersion,
+        CancellationToken cancellationToken = default)
+    {
+        if (!VersionUtils.TryParse(dataSetVersion, out var version))
+        {
+            return new NotFoundResult();
+        }
+
+        return await publicDataDbContext.DataSetVersions
+            .AsNoTracking()
+            .Include(dsv => dsv.DataSet)
+            .Where(dsv => dsv.DataSetId == dataSetId)
+            .Where(dsv => dsv.VersionMajor == version.Major)
+            .Where(dsv => dsv.VersionMinor == version.Minor)
+            .SingleOrNotFoundAsync(cancellationToken);
+    }
+
+    private async Task<Either<ActionResult, Unit>> CheckDataSetVersionBelongsToReleaseVersion(
+        DataSetVersion dataSetVersion,
+        ReleaseVersion releaseVersion)
+    {
+        return await contentDbContext.ReleaseFiles
+            .AsNoTracking()
+            .Where(rf => rf.ReleaseVersionId == releaseVersion.Id)
+            .Where(rf => rf.FileId == dataSetVersion.CsvFileId)
+            .SingleOrNotFoundAsync()
+            .OnSuccessVoid();
+    }
+
+    private DataSetVersionSummaryViewModel MapDataSetVersion(DataSetVersion dataSetVersion, ReleaseVersion releaseVersion)
+    {
+        return new DataSetVersionSummaryViewModel
+        {
+            Title = dataSetVersion.DataSet.Title,
+            Release = MapReleaseVersion(releaseVersion),
+            Version = dataSetVersion.Version,
+            Type = dataSetVersion.VersionType,
+            Status = dataSetVersion.Status,
+            DataSetFileId = dataSetVersion.CsvFileId,
+        };
+    }
+
+    private Release MapReleaseVersion(ReleaseVersion releaseVersion)
```

Comment:
This method could be made `static` which means `MapDataSetVersion` can then be made `static` too.